### DISCUSSION TITLE
Problems with Docker version of ARM

### DISCUSSION THREAD

**Discussion by HaroldRobi** [automatic-ripping-machine/automatic-ripping-machine#776] [Q&A]
I wanted to start a new discussion regarding the Docker version of ARM.  I have 4 DVD drives attached to my system.  All connected directly to individual USB ports on the computer.

The first two devices /dev/sr0 and /dev/sr1 appear to be working correctly.  They retrieve the poster of the DVD it is working on and rips what it needs to.

/dev/sr2 and /dev/sr3 both appear to not work.  Either the jobs will start then abort with no errors.  or the jobs will hang with the status "active" when hovering over the littler circle on the web page.  It will acknowledge that a DVD is present, create the appropriate log but nothing is done after that.   it never rips the DVD even though I let it go all night long.

At first I thought it was my USB hub, so I eliminated that, it now only supplies power to any drive that needs it.  the other reason I suspect something is wrong is because if I go to abort the job, it will not abort it.  It says "unable to abort job".

 An additional suggestion is if 4 drives are the maximum configured for a fresh docker installation, maybe it should be 6 instead.  That way a person does not have to fiddle around with the docker container to get all of their devices working properly.  Better yet the install script should ask how many drives you have in the system, then configure the docker automatically.

When installing the docker container, I did not have the other two drives plugged into the computer, I added those after I setup the container.  Would that cause a problem?

**Comment by 1337-server** [comment]
We need a few things to help can you share
- Your docker run command
- Logs from the jobs if available
- output of (inside the container) `ls -la /mnt/dev/`
- output of (inside the container) `cat /etc/fstab` 
- and the output from the container log when you insert 
You should be seeing something like 
`[ARM] Starting ARM for DVD/BD/CD on srx`
when you insert the device

Have you tried connecting to the container and running the manual command ? Replace x with drive number
`sudo -u arm /usr/bin/python3 /opt/arm/arm/ripper/main.py -d srx`
This will sometimes throw out an error that arm cant catch.

**Reply by HaroldRobi** [reply]
Being generally new to docker.  I have worked with docker under unraid.  Can I just ssh to the docker container from my windows pc .. or do I have to do that from the console. 

thanks again for all the help.

**Reply by HaroldRobi** [reply]
In response to your questions, see below:

- **Your docker run command ->** 
`docker run -d \`
    `-p "8080:8080" \`
    `-e ARM_UID="1001" \`
    `-e ARM_GID="1001" \`
    `-v "/home/arm:/home/arm" \`
    `-v "/home/arm/Music:/home/arm/Music" \`
    `-v "/home/arm/logs:/home/arm/logs" \`
    `-v "/home/arm/media:/home/arm/media" \`
    `-v "/etc/arm/config:/etc/arm/config" \`
    `--device="/dev/sr0:/dev/sr0" \`
    `--device="/dev/sr1:/dev/sr1" \`
    `--device="/dev/sr2:/dev/sr2" \`
    `--device="/dev/sr3:/dev/sr3" \`
    `--privileged \`
    `--restart "always" \`
    `--name "ripasaurus-2" \`
    `--cpuset-cpus='2,3,4,5' \`
    `automaticrippingmachine/automatic-ripping-machine:latest`

- **Logs from the jobs if available -> All job logs are empty**

- **output of (inside the container) ls -la /mnt/dev/**
`root@1f1433f7ec0e:/home/arm#  ls -la /mnt/dev/`
`total 44`
`drwxr-xr-x 11 root root 4096 Feb 25 01:07 .`
`drwxr-xr-x  1 root root 4096 Feb 25 01:07 ..`
`drwxrwxrwx  2 root root 4096 Feb 25 01:07 sr0`
`drwxrwxrwx  2 root root 4096 Feb 25 01:07 sr1`
`drwxrwxrwx  2 root root 4096 Feb 25 01:07 sr2`
`drwxrwxrwx  2 root root 4096 Feb 25 01:07 sr3`
`drwxrwxrwx  2 root root 4096 Feb 25 01:07 sr5`
`drwxrwxrwx  2 root root 4096 Feb 25 01:07 sr6`
`drwxrwxrwx  2 root root 4096 Feb 25 01:07 sr7`
`drwxrwxrwx  2 root root 4096 Feb 25 01:07 sr8`
`drwxrwxrwx  2 root root 4096 Feb 25 01:07 sr9`
`root@1f1433f7ec0e:/home/arm#`

- **output of (inside the container) cat /etc/fstab ->**
`root@1f1433f7ec0e:/home/arm# cat /etc/fstab`
`# UNCONFIGURED FSTAB FOR BASE SYSTEM`
`/dev/sr0  /mnt/dev/sr0  udf,iso9660  users,noauto,exec,utf8,ro  0  0`
`/dev/sr1  /mnt/dev/sr1  udf,iso9660  users,noauto,exec,utf8,ro  0  0`
`/dev/sr2  /mnt/dev/sr2  udf,iso9660  users,noauto,exec,utf8,ro  0  0`
`/dev/sr3  /mnt/dev/sr3  udf,iso9660  users,noauto,exec,utf8,ro  0  0`
`/dev/sr4  /mnt/dev/sr4  udf,iso9660  users,noauto,exec,utf8,ro  0  0`
`/dev/sr5  /mnt/dev/sr5  udf,iso9660  users,noauto,exec,utf8,ro  0  0`
`/dev/sr6  /mnt/dev/sr6  udf,iso9660  users,noauto,exec,utf8,ro  0  0`
`/dev/sr7  /mnt/dev/sr7  udf,iso9660  users,noauto,exec,utf8,ro  0  0`
`/dev/sr8  /mnt/dev/sr8  udf,iso9660  users,noauto,exec,utf8,ro  0  0`
`/dev/sr9  /mnt/dev/sr9  udf,iso9660  users,noauto,exec,utf8,ro  0  0`
`root@1f1433f7ec0e:/home/arm#`

- 'and the output from the container log when you insert'
'You should be seeing something like'
'[ARM] Starting ARM for DVD/BD/CD on srx'
'when you insert the device'

After some struggle this is what i see for the last 50 lines of the log

`hrobinson@ripasaurus-2:~$ sudo docker logs 1f1433f7ec0e -n 50`
`[2023-03-02 05:18:29,022] DEBUG ARM: json_api.get_x_jobs couldn't get config`
`[2023-03-02 05:18:29,022] DEBUG ARM: json_api.get_x_jobs jobs  - we have 3 jobs`
`[2023-03-02 05:18:31,013] DEBUG ARM: json_api.process_logfile ripping`
`[2023-03-02 05:18:31,013] DEBUG ARM: json_api.process_logfile using mkv -` `/home/arm/logs/LOONEY_TUNES_GOLDEN_DISC_1_167768401678.log`
`[2023-03-02 05:18:31,018] DEBUG ARM: json_api.process_logfile ripping`
`[2023-03-02 05:18:31,018] DEBUG ARM: json_api.process_logfile using mkv -` `/home/arm/logs/LOONEY_TUNES_GOLDEN_DISC_2_167768415455.log`
`[2023-03-02 05:18:31,022] DEBUG ARM: json_api.process_logfile active`
`[2023-03-02 05:18:31,022] DEBUG ARM: json_api.process_logfile using handbrake`
`[2023-03-02 05:18:31,024] DEBUG ARM: json_api.process_handbrake_logfile Cant find index`
`[2023-03-02 05:18:31,026] DEBUG ARM: json_api.get_x_jobs couldn't get config`
`[2023-03-02 05:18:31,026] DEBUG ARM: json_api.get_x_jobs jobs  - we have 3 jobs`
`[2023-03-02 05:18:33,016] DEBUG ARM: json_api.process_logfile ripping`
`[2023-03-02 05:18:33,016] DEBUG ARM: json_api.process_logfile using mkv -` `/home/arm/logs/LOONEY_TUNES_GOLDEN_DISC_1_167768401678.log`
`[2023-03-02 05:18:33,020] DEBUG ARM: json_api.process_logfile ripping`
`[2023-03-02 05:18:33,020] DEBUG ARM: json_api.process_logfile using mkv -` `/home/arm/logs/LOONEY_TUNES_GOLDEN_DISC_2_167768415455.log`
`[2023-03-02 05:18:33,025] DEBUG ARM: json_api.process_logfile active`
`[2023-03-02 05:18:33,025] DEBUG ARM: json_api.process_logfile using handbrake`
`[2023-03-02 05:18:33,027] DEBUG ARM: json_api.process_handbrake_logfile Cant find index`
`[2023-03-02 05:18:33,029] DEBUG ARM: json_api.get_x_jobs couldn't get config`
`[2023-03-02 05:18:33,029] DEBUG ARM: json_api.get_x_jobs jobs  - we have 3 jobs`
`[2023-03-02 05:18:35,017] DEBUG ARM: json_api.process_logfile ripping`
`[2023-03-02 05:18:35,017] DEBUG ARM: json_api.process_logfile using mkv -` `/home/arm/logs/LOONEY_TUNES_GOLDEN_DISC_1_167768401678.log`
`[2023-03-02 05:18:35,021] DEBUG ARM: json_api.process_logfile ripping`
`[2023-03-02 05:18:35,021] DEBUG ARM: json_api.process_logfile using mkv -` `/home/arm/logs/LOONEY_TUNES_GOLDEN_DISC_2_167768415455.log`
`[2023-03-02 05:18:35,026] DEBUG ARM: json_api.process_logfile active`
`[2023-03-02 05:18:35,026] DEBUG ARM: json_api.process_logfile using handbrake`
`[2023-03-02 05:18:35,028] DEBUG ARM: json_api.process_handbrake_logfile Cant find index`
`[2023-03-02 05:18:35,030] DEBUG ARM: json_api.get_x_jobs couldn't get config`
`[2023-03-02 05:18:35,030] DEBUG ARM: json_api.get_x_jobs jobs  - we have 3 jobs`
`[2023-03-02 05:18:37,004] DEBUG ARM: json_api.process_logfile ripping`
`[2023-03-02 05:18:37,004] DEBUG ARM: json_api.process_logfile using mkv -` `/home/arm/logs/LOONEY_TUNES_GOLDEN_DISC_1_167768401678.log`
`[2023-03-02 05:18:37,008] DEBUG ARM: json_api.process_logfile ripping`
`[2023-03-02 05:18:37,009] DEBUG ARM: json_api.process_logfile using mkv -` `/home/arm/logs/LOONEY_TUNES_GOLDEN_DISC_2_167768415455.log`
`[2023-03-02 05:18:37,013] DEBUG ARM: json_api.process_logfile active`
`[2023-03-02 05:18:37,013] DEBUG ARM: json_api.process_logfile using handbrake`
`[2023-03-02 05:18:37,016] DEBUG ARM: json_api.process_handbrake_logfile Cant find index`
`[2023-03-02 05:18:37,018] DEBUG ARM: json_api.get_x_jobs couldn't get config`
`[2023-03-02 05:18:37,018] DEBUG ARM: json_api.get_x_jobs jobs  - we have 3 jobs`
`[2023-03-02 05:19:05,374] DEBUG ARM: utils.arm_alembic_get Alembic Head is: 95623e8c5d58`
`[2023-03-02 05:19:05,376] DEBUG ARM: utils.arm_db_get Database Head is: 95623e8c5d58`
`[2023-03-02 05:19:05,376] DEBUG ARM: utils.arm_db_check Database is current. Head: 95623e8c5d58DB: 95623e8c5d58`
`[2023-03-02 05:19:36,015] DEBUG ARM: json_api.process_logfile ripping`
`[2023-03-02 05:19:36,015] DEBUG ARM: json_api.process_logfile using mkv -` `/home/arm/logs/LOONEY_TUNES_GOLDEN_DISC_1_167768401678.log`
`[2023-03-02 05:19:36,020] DEBUG ARM: json_api.process_logfile ripping`
`[2023-03-02 05:19:36,020] DEBUG ARM: json_api.process_logfile using mkv -` `/home/arm/logs/LOONEY_TUNES_GOLDEN_DISC_2_167768415455.log`
`[2023-03-02 05:19:36,024] DEBUG ARM: json_api.process_logfile active`
`[2023-03-02 05:19:36,024] DEBUG ARM: json_api.process_logfile using handbrake`
`[2023-03-02 05:19:36,027] DEBUG ARM: json_api.process_handbrake_logfile Cant find index`
`[2023-03-02 05:19:36,028] DEBUG ARM: json_api.get_x_jobs couldn't get config`
`[2023-03-02 05:19:36,028] DEBUG ARM: json_api.get_x_jobs jobs  - we have 3 jobs`
`hrobinson@ripasaurus-2:~$`

the "We have 3 jobs" is absolutely wrong.  No jobs were running at the time I inserted a DVD.  when I looked at the web interface, I could see 2 jobs running but I could not abandon them.  when I inserted the DVD the third job was created, and it can not be abandoned either. I am not using Handbrake, I am just ripping for the MKV file only.

**Reply by HaroldRobi** [reply]
I am following up on this question.

**Reply by 1337-server** [reply]
I can't see anything obvious in this. Everything looks good.

But I haven't had the experience with that many USB drives.
Does MakeMKV detect all of them ? `makemkvcon -r info` inside the container
Does your system detect them all ? `lsscsi -g` outside, on the host

**Reply by HaroldRobi** [reply]
> Does MakeMKV detect all of them ? makemkvcon -r info inside the container
`root@1f1433f7ec0e:/home/arm# makemkvcon -r info`
`MSG:1005,0,1,"MakeMKV v1.17.3 linux(x64-release) started","%1 started","MakeMKV v1.17.3 linux(x64-release)"`
`DRV:0,0,999,0,"BD-RE HL-DT-ST BD-RE BU40N 1.03 MZFL4T91101","","/dev/sr1"`
`DRV:1,2,999,1,"DVD+R-DL hp CDDVDW GP70N RR01 M08J43F3205","LOONEY_TUNES_GOLDEN_DISC_2","/dev/sr3"`
`DRV:2,1,999,0,"DVD+R-DL TSSTcorp CDDVDW SN-S082N LA00 SN-S082NFirmware","","/dev/sr2"`
`DRV:3,1,999,0,"BD-RE HL-DT-ST BD-RE BU40N 1.03 MOFM2QG0153","","/dev/sr0"`
`DRV:4,256,999,0,"","",""`
`DRV:5,256,999,0,"","",""`
`DRV:6,256,999,0,"","",""`
`DRV:7,256,999,0,"","",""`
`DRV:8,256,999,0,"","",""`
`DRV:9,256,999,0,"","",""`
`DRV:10,256,999,0,"","",""`
`DRV:11,256,999,0,"","",""`
`DRV:12,256,999,0,"","",""`
`DRV:13,256,999,0,"","",""`
`DRV:14,256,999,0,"","",""`
`DRV:15,256,999,0,"","",""`
`Use: makemkvcon [switches] Command [Parameters]`
``
`Commands:`
`  info <source>`
`      prints info about disc`
`  mkv <source> <title id> <destination folder>`
`      saves a single title to mkv file`
`  backup <source> <destination folder>`
`      backs up disc to a hard drive`
`  f <args>`
`      run universal firmware tool`
`  reg <key string or file name>`
`      enter registration key into program`

`Source specification:`
`  iso:<FileName>    - open iso image <FileName>`
`  file:<FolderName> - open files in folder <FolderName>`
`  disc:<DiscId>     - open disc with id <DiscId> (see list Command)`
`  dev:<DeviceName>  - open disc with OS device name <DeviceName>`

`Switches:`
` -r --robot        - turn on "robot" mode, see http://www.makemkv.com/developers`

> Does your system detect them all ? lsscsi -g outside, on the host

`<inside the container>`
`root@1f1433f7ec0e:/home/arm# lsblk`
`NAME        MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT`
`loop0         7:0    0    62M  1 loop`
`loop1         7:1    0  63.3M  1 loop`
`loop2         7:2    0  67.8M  1 loop`
`loop3         7:3    0  49.9M  1 loop`
`loop4         7:4    0  91.9M  1 loop`
`sr0          11:0    1  1024M  0 rom`
`sr1          11:1    1  1024M  0 rom`
`sr2          11:2    1  1024M  0 rom`
`sr3          11:3    1   5.7G  0 rom`
`nvme0n1     259:0    0 238.5G  0 disk`
`├─nvme0n1p1 259:1    0   1.1G  0 part`
`├─nvme0n1p2 259:2    0     2G  0 part`
`└─nvme0n1p3 259:3    0 235.4G  0 part`
`root@1f1433f7ec0e:/home/arm#`

`<Outside the container>`

`hrobinson@ripasaurus-2:~$ lsscsi -g`
`[2:0:0:0]    cd/dvd  TSSTcorp CDDVDW SN-S082N  LA00  /dev/sr2   /dev/sg2`
`[3:0:0:0]    cd/dvd  HL-DT-ST BD-RE BU40N      1.03  /dev/sr0   /dev/sg0`
`[4:0:0:0]    cd/dvd  HL-DT-ST BD-RE BU40N      1.03  /dev/sr1   /dev/sg1`
`[5:0:0:0]    cd/dvd  hp       CDDVDW GP70N     RR01  /dev/sr3   /dev/sg3`
`[N:0:0:1]    disk    KXG60ZNV256G KIOXIA__1                     /dev/nvme0n1`
`hrobinson@ripasaurus-2:~$`

Regards,

**Reply by HaroldRobi** [reply]
Just following up on my question, please let me know if I can be of further assistance.

**Reply by PriamX** [reply]
I'm not an ARM expert, but I do see a couple things that look odd to me, but I'm not sure if it's normal or not.

First, in your log output, I don't see any `INFO ARM` entries, only `DEBUG ARM` entries. This is odd, because both on my successful and failed rips there are plenty of `INFO ARM` entries. This is what I find most odd, that there's just not more log entries, but I'm not sure what that means.

Second, your optical drives, I know BU40N is a highly regarded and touted drive for use with MakeMKV. But I see neither the CP70N nor the SN-S082N drives on the recommended list for MakeMKV. I'm not sure if this means they won't work, have issues, or are just "okay" drives. Have you flashed any of the drives with MakeMKV's libredrive firmware? Or are they just stock firmware?

**Reply by HaroldRobi** [reply]
> First, in your log output, I don't see any INFO ARM entries, only DEBUG ARM entries. This is odd, because both on my successful and failed rips there are plenty of INFO ARM entries. This is what I find most odd, that there's just not more log entries, but I'm not sure what that means.

Well that was the last 50 lines of the log.  I went up to 200 lines back still all the same messages.

> Second, your optical drives, I know BU40N is a highly regarded and touted drive for use with MakeMKV. But I see neither the CP70N nor the SN-S082N drives on the recommended list for MakeMKV. I'm not sure if this means they won't work, have issues, or are just "okay" drives. Have you flashed any of the drives with MakeMKV's libredrive firmware? Or are they just stock firmware?

Both of the BU40Ns have been flashed using to the Libre drive firmware found at MakeMKV All these drives worked in an older install of ARM.  I moved these drives and decommissioned the first ARM I built in exchange of installing it on an HP 1L desktop I got for free.  The smaller PC is much faster than the other larger system I had.

**Reply by PriamX** [reply]
So the BU40Ns are currently working. The other two drives that aren't working, worked on a previous ARM...so hardware wouldn't seem to be the issue. makemkvcon detects the drives. Have you tried using makemkvcon to rip the disc(s) at the command line for the problem drives and see if it spits out any errors that aren't making it into the arm log file?

**Reply by shitwolfymakes** [reply]
Did you ever map the new devices in the run command? The container supports `/dev/sr0` through like `/dev/sr20`ish now, but they won't be utilized if they aren't mapped at container start to the host devices

**Reply by HaroldRobi** [reply]
> Did you ever map the new devices in the run command? The container supports /dev/sr0 through like /dev/sr20ish now, but they won't be utilized if they aren't mapped at container start to the host devices

I am not sure what you mean by "aren't mapped at container start to the host devices"  I have my Docker run command which was posted 5 days ago, is this what you are speaking of?  

Do all the drives need to be connected before I start the docker for the first time?

**Reply by 1337-server** [reply]
Before first starting, no. but you might need to restart the docker after you plug in the new drives.

**Reply by HaroldRobi** [reply]
Yes, I have restarted the docker and still have the same problem.  Only /dev/sr0 and /dev/sr1 are working none of the other drives drives are working properly.

**Reply by HaroldRobi** [reply]
I have decided to reinstall the server, I notced during the install of ARM for Docker I saw this:

`To run the Docker daemon as a fully privileged service, but granting non-root
users access, refer to https://docs.docker.com/go/daemon-access/

WARNING: Access to the remote API on a privileged Docker daemon is equivalent
         to root access on the host. Refer to the 'Docker daemon attack surface'
         documentation for details: https://docs.docker.com/go/attack-surface/

================================================================================

Pulling image from automaticrippingmachine/automatic-ripping-machine:latest
latest: Pulling from automaticrippingmachine/automatic-ripping-machine
7c3b88808835: Pull complete
6ea0e81b1046: Pull complete
2cdac9582559: Pull complete
265b6e77bea5: Pull complete
4f4fb700ef54: Pull complete
bd8cdda20c71: Pull complete
98e06d8c57bd: Pull complete
a18950ab2a16: Pull complete
f63620d206e7: Pull complete
4392922e7e9d: Pull complete
b5246add7aec: Pull complete
f66109487653: Pull complete
a9ee53e446eb: Pull complete
cb3535e788dd: Pull complete
d765249b2aa6: Pull complete
4c622b8edff1: Pull complete
423aaf7ca3a8: Pull complete
200b6656da80: Pull complete
74b408fc3c34: Pull complete
7c853e2d4cc4: Pull complete
1215e8b9d27d: Pull complete
18c9d9a6ff31: Pull complete
40910368c999: Pull complete
85ef86b096e8: Pull complete
de599021718d: Pull complete
6cf49ce325cc: Pull complete
477f8f3152a5: Pull complete
8956965884e5: Pull complete
0d5fbac49f42: Pull complete
47b261dd5277: Pull complete
59de5b6ddc8b: Pull complete
e045b9d44a34: Pull complete
1114d25b9965: Pull complete
2bc09a2fd1cc: Pull complete
Digest: sha256:83e69dcbaa7cdbf4cfcba0dd129d47155f935f185368402f36d5540f72dfe219
Status: Downloaded newer image for automaticrippingmachine/automatic-ripping-machine:latest
docker.io/automaticrippingmachine/automatic-ripping-machine:latest
Creating mount points
**./docker-setup.sh: line 94: /home/arm: Is a directory**
Installation complete. A template command to run the ARM container is located in:
hrobinson@ripasaurus-2:~$
`

Is the line bolded a concern?  Did the script crash?

**Reply by 1337-server** [reply]
You are best to set this up manually, it doesnt look like the setup script is able to be run multiple times.

Try using the docker run command instead to use the image.

Remeber that makemkv needs both locations for disc drives, so both sr0 and sg0 or what ever is output from `lsscsi -g` for each disc drive

https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/Docker-Troubleshooting#makemkv-cant-find-drives

**Reply by HaroldRobi** [reply]
So after the rebuild (I re-installed the OS then followed the directions on setting up the container.  I was able to get 3 drives to rip at the same time.  but inserting a 4th dvd to rip, causes ARM to reboot.  No kernel panic, as I was monitering the syslog using tail.  It just rebooted.  thinking this was a hardware issue, I grabed a lenovo 1 litre computer and install the newest version of ubuntu.  Installed the ARM container as per instructions, ran 3 drives with out issues.  Inserted the 4th DVD and after about a minute, It rebooted.  So I will just use 3 drives, which is better than 1 drive and 1 dvd at a time.

I am going to use samba, create a share point and have ARM save all rips there.  I will have to figure out how to make the connection between arm home folders and the samba share.

Also I am getting an error from ARM reporting about permissions on /dev/sr4

`ARM encountered a fatal error processing None. Check the logs for more details. [Errno 13] Permission denied: '/mnt/dev/sr4'`

This is what the log says:

`[04-06-2023 03:39:02] INFO ARM: Database is up to date`
`[04-06-2023 03:39:02] INFO ARM: ************* Starting ARM processing at 2023-04-06 03:39:02.368191 *************`
`[04-06-2023 03:39:03] INFO ARM: Branch: main - Commit: 01ac718`
`[04-06-2023 03:39:03] INFO ARM: ARM version: 2.6.27`
`[04-06-2023 03:39:03] INFO ARM: Python version: 3.8.10 (default, Mar 13 2023, 10:26:41) [GCC 9.4.0]`
`[04-06-2023 03:39:03] INFO ARM: User is: arm`
`[04-06-2023 03:39:03] INFO ARM: Looking for log files older than 1 days old.`
`[04-06-2023 03:39:03] INFO ARM: Checking path /home/arm/logs/ for old log files...`
`[04-06-2023 03:39:03] INFO ARM: Deleting log file: NIGHT_GALLERY.log`
`[04-06-2023 03:39:03] INFO ARM: Deleting log file: NIGHT_GALLERY_168058549049.log`
`[04-06-2023 03:39:03] INFO ARM: Deleting log file: NIGHT_GALLERY_168061125784.log`
`[04-06-2023 03:39:03] INFO ARM: Deleting log file: NIGHT_GALLERY_168058541895.log`
`[04-06-2023 03:39:03] INFO ARM: Deleting log file: NIGHT_GALLERY_168065677633.log`
`[04-06-2023 03:39:03] INFO ARM: Deleting log file: arm.log`
`[04-06-2023 03:39:03] INFO ARM: Deleting log file: NIGHT_GALLERY_168061130833.log`
`[04-06-2023 03:39:03] INFO ARM: Deleting log file: NIGHT_GALLERY_168065689679.log`
`[04-06-2023 03:39:03] INFO ARM: Checking path /home/arm/logs/progress for old log files...`
`[04-06-2023 03:39:03] INFO ARM: Deleting log file: 6.log`
`[04-06-2023 03:39:03] INFO ARM: Deleting log file: 9.log`
`[04-06-2023 03:39:03] INFO ARM: Deleting log file: 8.log`
`[04-06-2023 03:39:03] INFO ARM: Deleting log file: 10.log`
`[04-06-2023 03:39:03] INFO ARM: Deleting log file: 5.log`
`[04-06-2023 03:39:03] INFO ARM: Job: NIGHT_GALLERY`
`[04-06-2023 03:39:03] INFO ARM: Job #15 with PID 28309 is currently running.`
`[04-06-2023 03:39:03] INFO ARM: Starting Disc identification`
`[04-06-2023 03:39:03] INFO ARM: Mounting disc to: /mnt/dev/sr4`
`[04-06-2023 03:39:03] ERROR ARM: A fatal error has occurred and ARM is exiting.  See traceback below for details.`

Directory permissions in /mnt:

`hrobinson@ripasaurus-2:/mnt/dev$ ls -alh`
`total 28K`
`drwxr-xr-x 7 root root 4.0K Apr  1 23:35 .`
`drwxr-xr-x 3 root root 4.0K Apr  1 23:35 ..`
`drwxr-xr-x 2 arm  arm  4.0K Apr  1 23:35 sr0`
`drwxr-xr-x 2 arm  arm  4.0K Apr  1 23:35 sr1`
`drwxr-xr-x 2 arm  arm  4.0K Apr  1 23:35 sr2`
`drwxr-xr-x 2 arm  arm  4.0K Apr  1 23:35 sr3`
`drwxr-xr-x 2 arm  arm  4.0K Apr  1 23:35 sr4`
`hrobinson@ripasaurus-2:/mnt/dev$`

**Reply by HaroldRobi** [reply]
Here is an update to my issue.

I have moved ARM to a different 1 liter computer.  I am now using a Lenovo M700 with 8GB ram, Currently running Ubuntu 22.10.  It should not make a difference because of docker.  I insalled docker as per the instructions provided, and started testing.  I worked my way up to ripping 3 disks at once which was successful, I attempted to run a 4th disk but it would eject the disk almost immediately with a permission denied error.

I checked /dev/mnt on the local machine, and found all the sr# present. (sr0-sr4).  I checked permissions on those directories on the local machine and those were also correct.  I let the installation sit a while (life you know) then started working on it tonight.  I logged into the docker image using:  `sudo docker exec -it 645c9a546e27 sh.`  I went to the /dev/mnt folder.  found sr0-sr20 created there.  The only one missing was sr4.  I created that folder and made sure the permissions were set to 777, just like the others.  I inserted a dvd into /dev/sr4.  It mounted and started ripping the dvd.  It is currently ripping all 4 disks at the same time and it has not rebooted yet. So I suspect I had a hardware issue with the HP unit I was using.

I will post here and let everyone know how I got along.
Here it is working away at the current batch of DVD's  The bottom drive is a DVD drive I beleive is actually broken.  As it will not come ready when a dvd is inserted.

![PXL_20230409_063020942](https://user-images.githubusercontent.com/84169293/230758228-27772e9c-ae09-4df8-86eb-80233c6b23d8.jpg)
